File: Project/Java/src/eshop/Pencil.java
package eshop;

public class Pencil extends Item{
    public static final int HB = 0;
    public static final int H = 1;
    public static final int B = 2;
    private int type;
    private double tipSize;
    private static int count = 0;


    Pencil(String name, double price, String description, int stock, int id, double tipSize, int type){
        setName(name);
        setPrice(price);
        setDescription(description);
        setStock(stock);
        this.type = type;
        this.tipSize = tipSize;
        count++;
    }
    public String getDetails(){
       return "Type: " + type + ", Tip size: " + tipSize;
    }
    public static int getCount() {
        return count;
    }

}

File: Project/C++/Project1/Pencil.cpp
#include "Pencil.h"
int Pencil::ammount = 0;
Pencil::Pencil(std::string name, double price, std::string description, int stock, int id, double tipSize, int type)
{
	setName(name);
	setDescription(description);
	setPrice(price);
	setStock(stock);
	setId(id);
	setObjType("pencil");
	this->tipSize = tipSize;
	this->type = type;
	this->ammount++;
}

std::string Pencil::getDetails()
{
	std::string det = "Tip Size: " + std::to_string(tipSize) + ". Type: ";
	if (type == 0) {
		det += "HB.";
	}
	else if (type == 1) {
		det += "H.";
	}
	else {
		det += "B.";
	}
	return det;
}
